service/history/queues/executable.go

```diff
@@ -516,3 +519,19 @@ func (e *executableImpl) estimateTaskMetricTag() []metrics.Tag {
 		metrics.OperationTag(taskType), // for backward compatibility
 	}
 }
+
+func (e *executableImpl) capturePanic(retError *error) {
+	if panicObj := recover(); panicObj != nil {
```

Comment:
This **_IS_** called inside a deferred function.

Reply:
This require the capturePanic be called directly by a defer, like 
defer capturePanic(&retErr)
How do we prevent people do this:
defer func() { capturePanic(&retErr) }

Reply:
I think now we have this linter that can help capture this issue? Or I can get rid of this function and embed it in Execute(), I don't think this function will/should be called in other places.